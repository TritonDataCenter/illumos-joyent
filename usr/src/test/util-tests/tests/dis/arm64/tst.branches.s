/*
 * This file and its contents are supplied under the terms of the
 * Common Development and Distribution License ("CDDL"), version 1.0.
 * You may only use this file in accordance with the terms of version
 * 1.0 of the CDDL.
 *
 * A full copy of the text of the CDDL should have accompanied this
 * source.  A copy of the CDDL is also available via the Internet at
 * http://www.illumos.org/license/CDDL.
 */

/*
 * Copyright 2018 Joyent, Inc.
 */

/*
 * Test Data - Immediate group dissassembly
 */

.text
.align 16
.globl libdis_test
.type libdis_test, %function
libdis_test:
	/* Branch (conditional) */
	B.EQ branch_target
	B.NE branch_target
	B.CS branch_target
	B.CC branch_target
	B.MI branch_target
	B.PL branch_target
	B.VS branch_target
	B.VC branch_target
	B.HI branch_target
	B.LS branch_target
	B.GE branch_target
	B.LT branch_target
	B.GT branch_target
	B.LE branch_target
	B.AL branch_target
	B.NV branch_target

	/* Unconditional Branch (register) */
	BR X0
	BR XZR
	BLR X19
	RET /* Defaults to X30 */
	RET X9
	ERET
	DRPS

	/* Unconditional Branch (immediate) */
	B branch_target
	BL branch_target

	/* Compare and Branch (immediate) */
	CBZ W19, branch_target
	CBZ XZR, branch_target
	CBNZ W3, branch_target
	CBNZ X2, branch_target

	/* Test and Branch (immediate) */
	TBZ WZR, #19, branch_target
	TBZ X24, #35, branch_target
	TBNZ W4, #2, branch_target
	TBNZ X16, #62, branch_target

	branch_target: ADD X0, X0, X0
.size libdis_test, [.-libdis_test]

