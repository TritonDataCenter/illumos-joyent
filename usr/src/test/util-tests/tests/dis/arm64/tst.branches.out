    libdis_test:      20 04 00 54  b.eq +0x84 	<libdis_test+0x84>
    libdis_test+0x4:  01 04 00 54  b.ne +0x80 	<libdis_test+0x84>
    libdis_test+0x8:  e2 03 00 54  b.cs +0x7c 	<libdis_test+0x84>
    libdis_test+0xc:  c3 03 00 54  b.cc +0x78 	<libdis_test+0x84>
    libdis_test+0x10: a4 03 00 54  b.mi +0x74 	<libdis_test+0x84>
    libdis_test+0x14: 85 03 00 54  b.pl +0x70 	<libdis_test+0x84>
    libdis_test+0x18: 66 03 00 54  b.vs +0x6c 	<libdis_test+0x84>
    libdis_test+0x1c: 47 03 00 54  b.vc +0x68 	<libdis_test+0x84>
    libdis_test+0x20: 28 03 00 54  b.hi +0x64 	<libdis_test+0x84>
    libdis_test+0x24: 09 03 00 54  b.ls +0x60 	<libdis_test+0x84>
    libdis_test+0x28: ea 02 00 54  b.ge +0x5c 	<libdis_test+0x84>
    libdis_test+0x2c: cb 02 00 54  b.lt +0x58 	<libdis_test+0x84>
    libdis_test+0x30: ac 02 00 54  b.gt +0x54 	<libdis_test+0x84>
    libdis_test+0x34: 8d 02 00 54  b.le +0x50 	<libdis_test+0x84>
    libdis_test+0x38: 6e 02 00 54  b.al +0x4c 	<libdis_test+0x84>
    libdis_test+0x3c: 4f 02 00 54  b.nv +0x48 	<libdis_test+0x84>
    libdis_test+0x40: 00 00 1f d6  br x0
    libdis_test+0x44: e0 03 1f d6  br xzr
    libdis_test+0x48: 60 02 3f d6  blr x19
    libdis_test+0x4c: c0 03 5f d6  ret
    libdis_test+0x50: 20 01 5f d6  ret x9
    libdis_test+0x54: e0 03 9f d6  eret
    libdis_test+0x58: e0 03 bf d6  drps
    libdis_test+0x5c: 0a 00 00 14  b +0x28 	<libdis_test+0x84>
    libdis_test+0x60: 09 00 00 94  bl +0x24 	<libdis_test+0x84>
    libdis_test+0x64: 13 01 00 34  cbz w19, +0x20 	<libdis_test+0x84>
    libdis_test+0x68: ff 00 00 b4  cbz xzr, +0x1c 	<libdis_test+0x84>
    libdis_test+0x6c: c3 00 00 35  cbnz w3, +0x18 	<libdis_test+0x84>
    libdis_test+0x70: a2 00 00 b5  cbnz x2, +0x14 	<libdis_test+0x84>
    libdis_test+0x74: 9f 00 98 36  tbz wzr, #0x13, +0x10 	<libdis_test+0x84>
    libdis_test+0x78: 78 00 18 b6  tbz x24, #0x23, +0xc 	<libdis_test+0x84>
    libdis_test+0x7c: 44 00 10 37  tbnz w4, #0x2, +0x8 	<libdis_test+0x84>
    libdis_test+0x80: 30 00 f0 b7  tbnz x16, #0x3e, +0x4 	<libdis_test+0x84>
    libdis_test+0x84: 00 00 00 8b  add x0, x0, x0
